Units scaling and multiple timezones

ktsaou opened the pull request:
fixes #2745
fixes #2698 

This PR adds:

1. auto-scaling of units at the dashboard
2. units-conversions (e.g. temperatures)
3. time zones conversions at the dashboard

## new dashboard settings

![screenshot from 2017-10-10 23-54-01](https://user-images.githubusercontent.com/2662304/31409982-587d96f4-ae16-11e7-8cab-7b66d6b0eb6e.png)

---

The dashboard does dynamic units conversion, on the fly, like the following. Note that the chart units change, based on the value of the selected dimension:

![peek 2017-10-06 22-58](https://user-images.githubusercontent.com/2662304/31296227-309a4c06-aaea-11e7-8c7b-3bca6a954372.gif)

---

With dynamic unit conversion, certain cases may seem awkward. Like this:

![image](https://user-images.githubusercontent.com/2662304/31255526-f8f9ea0e-aa35-11e7-8bfa-d1066afaf311.png)

Custom dashboards can add `data-common-units="NAME"`. All the charts with the same `NAME` will get the same units.

---

Tests to be done:

- [x] test it works for both auto-scaling and original.
- [x] test it works when switching / enabling / disabling dimensions at the legend.
- [x] test it works when zooming in / out the charts.
- [x] test it works when hovering over the chart, after zooming in/out and enabling / disabling dimensions.
- [x] test custom dashboards can overwrite this per chart (`data-desired-units="auto"` or `data-desired-units="original"`).
- [x] test custom dashboards can give their desired scale (`data-desired-units="GB"`, when the chart is by default on `MB`).
- [x] support `data-common-units="NAME"`
- [x] test how it interacts with existing dashboard that have overwritten the units with `data-units="UNITS"`.
- [x] support converting Celsius to Fahrenheit. Another setting should be introduced to control the desired temperature units (or even a new tab called `Units` that all unit conversions should be maintained at).
- [x] support time zone conversions

Rockstar04 commented:
I just applied this to a few of my servers and it looks and works great. I don't have any custom dashboards, so I can't easily help check off those last boxes though.

ktsaou commented:
@Rockstar04 thanks!
Did netdata detect the right time zone of your servers properly? Which distros did you check?

Ferroin commented:
Doesn't appear to break anything on 64-bit Gentoo, but I didn't check the timezone stuff thoroughly. It appears to have correctly detected the time zones on my servers, but they're all set to the same time zone, and I didn't check with a browser on a system using a different time zone.

Rockstar04 commented:
Times and timezones were all correct. All of our servers are 14.04/16.04 Ubuntu and a single  Debian Stretch server.

https://imgur.com/ciHTvDc

ktsaou commented:
@Rockstar04 what I am interested is the 2 timezones presented here:

![image](https://user-images.githubusercontent.com/2662304/31353256-7ed1328e-ad3a-11e7-9c42-c0efb34a614a.png)

Especially the `server reported timezone`.

Rockstar04 commented:
![capture](https://user-images.githubusercontent.com/641721/31386187-855238aa-ad8b-11e7-822e-8641bb100e0d.PNG)

America/Chicago is our correct timezone. Side note, is [globa] supposed to be [global] ?

Ferroin commented:
Did some more checking, and I'm seeing correct times and timezones, even when setting random timezones on the client system.

ktsaou commented:
> America/Chicago is our correct timezone. Side note, is [globa] supposed to be [global] ?

yeap, that is a typo...

ktsaou commented:
completed the tests.
everything seems good.

ktsaou commented:
![image](https://user-images.githubusercontent.com/2662304/31411264-c1e68454-ae19-11e7-9870-2bd633443545.png)

![image](https://user-images.githubusercontent.com/2662304/31411328-ef7558a0-ae19-11e7-9905-dbb95b261d12.png)

ktsaou commented:
![image](https://user-images.githubusercontent.com/2662304/31411264-c1e68454-ae19-11e7-9870-2bd633443545.png)

![image](https://user-images.githubusercontent.com/2662304/31411328-ef7558a0-ae19-11e7-9905-dbb95b261d12.png)

sagar-chandarana commented:
@ktsaou please update wiki with some examples at  https://github.com/firehol/netdata/wiki/Custom-Dashboards

ktsaou commented:
about settings related to units scaling on custom dashboards?

sagar-chandarana commented:
Yes exactly. `data-desired-units`

On Feb 26, 2018 20:28, "Costa Tsaousis" <notifications@github.com> wrote:

> about settings related to units scaling on custom dashboards?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/firehol/netdata/pull/2847#issuecomment-368688630>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADXEuZdzWoQQq_fTQJ_ObM8zCs2XLBaWks5tYz4ngaJpZM4Pvyzx>
> .
>

ktsaou commented:
ok, added a lot. This still far from complete...
I also added links to the code that shows per chart-library settings.

If anyone is willing to contribute some documentation in this area, please do so...
